src/publicodes-state/helpers/safeGetSituation.ts

```diff
@@ -1,18 +1,53 @@
-import { NodeValue, Situation } from '../types'
+import { Situation } from '../types'
+import { dottedNamesMigration } from './dottedNamesMigration'
 
 export const safeGetSituation = ({
   situation,
   everyRules,
 }: {
   situation: Situation
   everyRules: string[]
-}): any =>
-  everyRules
-    .filter((rule: string) => situation[rule] || situation[rule] === 0)
-    .reduce(
-      (accumulator: Record<string, NodeValue>, currentValue: string) => ({
-        ...accumulator,
-        [currentValue]: situation[currentValue],
-      }),
-      {}
-    )
+}): any => {
+  const unsupportedDottedNamesFromSituation = Object.keys(situation).filter(
+    (dottedName) => {
+      // We check if the dotteName is a rule of the model and migrate the rule if it's possible
+      if (!everyRules.includes(dottedName)) {
+        return true
+      }
+      // We check if the value of a dottedName waiting for an answer from "une possiblité"
+      // is defined as a rule `dottedName . value` in the model
+      // and migrate the rule if it's possible
+      if (
+        typeof situation[dottedName] === 'string' &&
+        situation[dottedName] !== 'oui' &&
+        situation[dottedName] !== 'non' &&
+        !everyRules.includes(`${dottedName} . ${situation[dottedName]}`)
+      ) {
+        return true
+      }
+    }
+  )
+
+  const filteredSituation = situation
```

Comment:
```suggestion
  const filteredSituation = {...situation} 
```
Évite des emmer**s potentielles

Reply:
Sans ça filteredSituation est simplement une référence à l'objet situation passé à la fonction, en le modifiant tu peux déclencher des effets de bord involontaires.

Reply:
Attention tout de même parce que le spread operator ne va pas faire une copie profonde de ton objet (cf. https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy)

Reply:
Hmm ça pourrait poser problème dans ce cas-là tu penses ? 🤔 je sais pas du tout

Reply:
Ça posait posait bien pb avec la volonté de conserver l'ancien localstorage en plus de celui migré, bien vu milou

Reply:
C'est pas ouf non plus mais j'ai trouvé qu'on pouvait utiliser `JSON.parse(JSON.stringify(object))` mais je crois que c'est pas optimal non plus